src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ReplaceElementAttributesWithNeighborValues.cpp

```diff
@@ -0,0 +1,267 @@
+  LessThanComparison& operator=(LessThanComparison&&) = delete;      // Move Assignment Not Implemented
+
+  bool compare(T a, T b) const override
+  {
+    return a < b;
+  }
+  bool compare1(T a, T b) const override
+  {
+    return a >= b;
+  }
+  bool compare2(T a, T b) const override
+  {
+    return a > b;
+  }
+};
+
+template <typename T>
+class GreaterThanComparison : public IComparisonFunctor<T>
+{
+public:
+  GreaterThanComparison() = default;
+  ~GreaterThanComparison() override = default;
+  GreaterThanComparison(const GreaterThanComparison&) = delete;            // Copy Constructor Not Implemented
+  GreaterThanComparison(GreaterThanComparison&&) = delete;                 // Move Constructor Not Implemented
+  GreaterThanComparison& operator=(const GreaterThanComparison&) = delete; // Copy Assignment Not Implemented
+  GreaterThanComparison& operator=(GreaterThanComparison&&) = delete;      // Move Assignment Not Implemented
+
+  bool compare(T a, T b) const override
+  {
+    return a > b;
+  }
+  bool compare1(T a, T b) const override
+  {
+    return a <= b;
+  }
+  bool compare2(T a, T b) const override
+  {
+    return a < b;
+  }
+};
+
+struct ExecuteTemplate
+{
+
+  template <typename T>
+  void operator()(const DataStructure& dataStructure, const ImageGeom& imageGeom, IDataArray& inputIDataArray, int32 comparisonAlgorithm, float thresholdValue, bool loopUntilDone,
+                  const std::atomic_bool& shouldCancel, const IFilter::MessageHandler& messageHandler)
+  {
+    using DataArrayType = DataArray<T>;
+
+    const DataArrayType& inputArray = dynamic_cast<const DataArrayType&>(inputIDataArray);
+
+    size_t totalPoints = inputArray.getNumberOfTuples();
+
+    Vec3 udims = imageGeom.getDimensions();
+    int64_t dims[3] = {
+        static_cast<int64_t>(udims[0]),
+        static_cast<int64_t>(udims[1]),
+        static_cast<int64_t>(udims[2]),
+    };
+
+    bool good = true;
+    int64_t neighbor = 0;
+    int64_t column = 0, row = 0, plane = 0;
+
+    int64_t neighborPoints[6] = {0, 0, 0, 0, 0, 0};
+    neighborPoints[0] = static_cast<int64_t>(-dims[0] * dims[1]);
+    neighborPoints[1] = static_cast<int64_t>(-dims[0]);
+    neighborPoints[2] = static_cast<int64_t>(-1);
+    neighborPoints[3] = static_cast<int64_t>(1);
+    neighborPoints[4] = static_cast<int64_t>(dims[0]);
+    neighborPoints[5] = static_cast<int64_t>(dims[0] * dims[1]);
+
+    std::vector<int64_t> bestNeighbor(totalPoints, -1);
+
+    size_t count = 0;
+    float best = 0.0f;
+    bool keepGoing = true;
+
+    std::shared_ptr<IComparisonFunctor<T>> comparator = std::make_shared<LessThanComparison<T>>();
+    if(comparisonAlgorithm == k_GreaterThan)
+    {
+      comparator = std::make_shared<GreaterThanComparison<T>>();
+    }
+
+    const AttributeMatrix* attrMatrix = imageGeom.getCellData();
+
+    for(const auto& [dataId, dataObject] : *attrMatrix)
+    {
+    }
+    while(keepGoing)
+    {
+      keepGoing = false;
```

Comment:
```suggestion
    do {
```
instead of a pseudo while true while(keepgoing), one should use a do-while with the loopUntilDone && count > 0 as the while conditon

Reply:
https://stackoverflow.com/questions/994905/is-there-ever-a-need-for-a-do-while-loop